### PR TITLE
More generic fix for HuggingFace LayerNorm.weight reference - replacing with PyTorch equivalents

### PR DESCRIPTION
**Pull request by michaellavelle**
I read on the README that "mingpt/trainer.py is (GPT-independent) PyTorch boilerplate" and realised the previous fix (https://github.com/karpathy/minGPT/pull/16 ) for the layer norm weight exclusion was coupled to the GPT structure.

This PR should resolve that by allowing a set of module types to be configured whose parameters are to be excluded from weight decay.

### CODE REVIEW

**Comment by michaellavelle** [comment]
Update - a new PR which also excludes the positional and token embeddings from weight decay:  https://github.com/karpathy/minGPT/pull/24

**Comment by karpathy** [comment]
fixed in https://github.com/karpathy/minGPT/commit/bbbdac74fa9b2e55574d70056163ffbae42310c1
